aspnetcore/mvc/views/tag-helpers/built-in/AnchorTagHelper.md

```diff
@@ -34,35 +32,110 @@ The speaker controller used in attribute definitions below is shown here.
+`asp-controller` is used to associate which controller will be used to generate the URL. The controllers specified must exist in the current project. The following code lists all speakers: 
+
+```
+<a asp-controller="Speaker" asp-action="Index" >All Speakers</a>
+```
+
+The generated URL will be
+
+```
+<a href="/Speaker">All Speakers</a>
+```
+
+If the `asp-controller` is specified and `asp-action` is not, the default `asp-action` will be the default controller method of the currently executing view.  That is, in the above example, if `asp-action` is left out, and this Anchor Tag Helper generated from the Home index view (**/Home**), the generated markup will be:
+
+
+```html
+<a href="/Home">All Speakers</a>
+```
 
 - - -
   
 ### asp-action
 
-`asp-action` is the name of the method in the controller that will be included in the final URL. That is, in the example, if the route to the Speaker Detail page is wanted, then the attribute should be set to `asp-action=Detail`. You should always set `asp-controller` when specifying `asp-action`. If no `asp-action` is specified then the default `asp-controller` will be the current executing controller.
+`asp-action` is the name of the action method in the controller that will be included in the generated `href`. For example, the following code set the generated `href` to point to the speaker detail page:
+
+```html
+<a asp-controller="Speaker" asp-action="Detail" >Speaker Detail</a>
+```
+
+The generated URL will be
+
+```html
+<a href="/Speaker/Detail">Speaker Detail</a>
+```
+
+If no `asp-controller` attribute is specified, the default controller calling the view executing the current view will be used.  
+ 
+If the attribute `asp-action` is `Index`, then no action is appended to the URL, leading to the default `Index` method being called. The action specified (or defaulted), must exist in the controller referenced in `asp-controller`.
 
 - - -
   
 ### asp-route-{value}
 
-`asp-route-` is a wild card route prefix. Any value you put after the trailing dash will be interpreted as the parameter to pass into the route. For example, if a tag is created as follows: 
+`asp-route-` is a wild card route prefix. Any value you put after the trailing dash will be interpreted as a potential route parameter. If a default route is not found, this route prefix will be appended to the generated href as a request parameter and value.  Otherwise it will be substituted in the route template.
+
+Assuming you have a controller method defined as follows:
+
+```csharp
+public IActionResult AnchorTagHelper(string id)
+{
+    var speaker = new SpeakerData()
+    {
+        SpeakerId = 12
+    };      
+    return View(viewName, speaker);
+}
+```
 
-`<a  asp-controller="Speaker" asp-action="Detail" asp-route-id-="11">Speaker 11</a>`
+and have the default route template defined in your **startup.cs** as follows:
```

Comment:
Files are italic, please check all `**` and make sure you have the right styling.